plugins/extractors/clickhouse/clickhouse.go

```diff
@@ -53,13 +55,27 @@ func New(logger log.Logger) *Extractor {
 }
 
 // Init initializes the extractor
-func (e *Extractor) Init(ctx context.Context, config plugins.Config) error {
-	if err := e.BaseExtractor.Init(ctx, config); err != nil {
+func (e *Extractor) Init(ctx context.Context, config plugins.Config) (err error) {
+	err = e.BaseExtractor.Init(ctx, config)
+	if err != nil {
 		return err
 	}
 
-	var err error
-	e.db, err = sql.Open("clickhouse", e.config.ConnectionURL)
+	driverName := "clickhouse"
+	if config.OtelEnabled {
+		driverName, err = otelsql.Register(driverName,
+			otelsql.AllowRoot(),
+			otelsql.TraceQueryWithoutArgs(),
+			otelsql.TraceRowsClose(),
+			otelsql.TraceRowsAffected(),
+			otelsql.WithSystem(semconv.DBSystemClickhouse),
+		)
+		if err != nil {
+			return fmt.Errorf("register clickhouse otelsql wrapper: %w", err)
+		}
+	}
+
+	e.db, err = sql.Open(driverName, e.config.ConnectionURL)
```

Comment:
We should also record stats for the opened database:

https://github.com/nhatthm/otelsql#usage

> Optionally, you could record [database connection metrics](https://github.com/nhatthm/otelsql#database-connection-metrics) using the `otelsql.RecordStats()`. For example:

> ```go
>	// Connect to a Postgres database using the postgres driver wrapper.
>	db, err := sql.Open(driverName, dsn)
>	if err != nil {
>		return nil, err
>	}
>
>	if err := otelsql.RecordStats(db); err != nil {
>		return nil, err
>	}
>
>	return db, nil
> ```

Reply:
updated